format-common/src/main/java/io/cdap/plugin/format/plugin/AbstractFileSink.java

```diff
@@ -64,10 +68,25 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     // invalid
     collector.getOrThrowException();
 
+    if (config.containsMacro(NAME_FORMAT)) {
+      // Deploy all format plugins. This ensures that the required plugin is available when
+      // the format macro is evaluated in prepareRun,
+      for (FileFormat f: FileFormat.values()) {
+        try {
+          pipelineConfigurer.usePlugin(ValidatingOutputFormat.PLUGIN_TYPE, f.name().toLowerCase(),
+                                       f.name().toLowerCase(), config.getRawProperties());
+        } catch (InvalidPluginConfigException e) {
+          LOG.warn("Failed to configure {}, missing properties: {}, invalid properties: {}",
```

Comment:
error messages should inform the reader about the implications. Something like:

Failed to register format '{}', which means it cannot be used when the pipeline is run. Missing properties ...